libsolidity/analysis/TypeChecker.cpp

```diff
@@ -716,7 +717,10 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 	// type is filled either by ReferencesResolver directly from the type name or by
 	// TypeChecker at the VariableDeclarationStatement level.
 	TypePointer varType = _variable.annotation().type;
-	solAssert(!!varType, "Failed to infer variable type.");
+	if (!varType)
```

Comment:
Can you explain why we cannot use the assertion here? This looks rather dangerous to me...